### PR TITLE
[Bugfix]Fix the problem of "," being divided in []

### LINKED ISSUE #5400: [Bug] [Seatunnel] Seatunnel -i command bug
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.

### What happened

image

image

### SeaTunnel Version

2.3.3

### SeaTunnel Config

```conf
env {
"job.name"="SeaTunnel_Job"
"job.mode"=BATCH
}
source {
Jdbc {
    "partition_upper_bound"="${read_para}"
}
}
transform {
}
sink {
Jdbc {
    
}
}
```

### Running Command

```shell
seatunnel.sh --config /tmp/dolphinscheduler/exec/syncJobTmpConfig/51_10734653065024.config --async -n test-hive -i read_para=[par1=20230829,par2=20230829]
```

### Error Exception

```log
When using JCommander, the input [par1=20230829,par2=20230829] is being parsed into two separate strings instead of being treated as a single unit.
```

### Zeta or Flink or Spark Version

_No response_

### Java or Scala Version

java 1.8

### Screenshots

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by liugddx**
## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: seatunnel-core/seatunnel-core-starter/src/main/java/org/apache/seatunnel/core/starter/command/ParameterSplitter.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        while (matcher.find()) {
+            stringBuilder.append(value, start, matcher.start());
+            if (matcher.group().equals(",")) {
+                stringBuilder.append(";");
+            } else {
+                stringBuilder.append(matcher.group());
+            }
+            start = matcher.end();
+        }
+        stringBuilder.append(value.substring(start)); // 添加剩余部分
```
Comment (ruanwenjun) [inline seatunnel-core/seatunnel-core-starter/src/main/java/org/apache/seatunnel/core/starter/command/ParameterSplitter.java]:
```suggestion
        stringBuilder.append(value.substring(start)); 
```

File: seatunnel-core/seatunnel-core-starter/src/main/java/org/apache/seatunnel/core/starter/command/ParameterSplitter.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class ParameterSplitter implements IParameterSplitter {
+
+    Pattern pattern = Pattern.compile("\\[.*?]|,");
+
+    @Override
+    public List<String> split(String value) {
```
Comment (ruanwenjun) [inline seatunnel-core/seatunnel-core-starter/src/main/java/org/apache/seatunnel/core/starter/command/ParameterSplitter.java:28]: Please add doc and UT for this method, and if the params contains ";" what will happen?
e.g. if the args looks like -i "map = ["par1=20230829;", "par2=20230829;"]"

Comment (liugddx) [reply seatunnel-core/seatunnel-core-starter/src/main/java/org/apache/seatunnel/core/starter/command/ParameterSplitter.java:28]: > Please add doc and UT for this method, and if the params contains ";" what will happen? e.g. if the args looks like -i "map = ["par1=20230829;", "par2=20230829;"]"

Yes, this code has a bug, I modified the implementation.PTAL

**Review by ruanwenjun — DISMISSED** [review]
Basically LGTM.

I am not sure if there still exist some cases will have another problem, e.g "-i map=["par1=[2023, 2024, 2025]", "par2=[2026, 2027]"]".

It might be better to use some inner method provided by JCommander to handle the array.

**Comment by liugddx** [comment]
Since the quotes [] cannot be parsed, special processing is required.
93c6163773f0e38a7857eb7cf55fb5b

**Comment by liugddx** [comment]
@ruanwenjun @EricJoy2048 PTAL

**Review by EricJoy2048 — APPROVED** [review]

**Review by ruanwenjun — APPROVED** [review]
+1

**Pull request merged**